### PR TITLE
feat: APP-412 add NotFoundPage when no project is found

### PR DESCRIPTION
**Pull request by r41ph**
## Description

https://regennetwork.atlassian.net/browse/APP-412
---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. Go to the marketplace and enter a random `slug/projectId` where the project Id goes in the URL (`https://deploy-preview-2521--regen-marketplace.netlify.app/project/PROJECT_ID_OR_SLUG_GOES_HERE/buy`, i.e: `https://deploy-preview-2521--regen-marketplace.netlify.app/project/xxxxx/buy`) and verify that the app redirects to the 404 page instead of the Buy credits flow.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by r41ph** [comment]
@erikalogie see testing instructions

**Comment by erikalogie** [comment]
I'm still getting the buy credits flow
Screenshot 2024-10-28 at 4 27 44 PM

**Comment by blushi** [comment]
> I'm still getting the buy credits flow Screenshot 2024-10-28 at 4 27 44 PM

@r41ph looks like you didn't provide the right deploy preview url in the testing instructions, it should be tested on https://deploy-preview-2521--regen-marketplace.netlify.app/

File: web-marketplace/src/pages/BuyCredits/BuyCredits.tsx [hunk 1]
```diff
@@ -80,6 +86,16 @@ export const BuyCredits = () => {
     if (confirmationTokenId) summarizePayment(confirmationTokenId);
   }, [confirmationTokenId, summarizePayment]);
 
+  const noProjectFound =
+    !loadingProjectByOnChainId &&
+    !projectByOnChainId &&
+    !loadingProjectBySlug &&
+    !projectBySlug?.data?.projectBySlug &&
+    !loadingOffchainProjectById &&
+    !offchainProjectByIdData;
```
Comment (blushi) [inline web-marketplace/src/pages/BuyCredits/BuyCredits.tsx:95]: should we use the exact same condition as in ProjectDetails and try to refactor this a bit to have `noProjectFound` returned by `useGetProject`

**Comment by r41ph** [comment]
> I'm still getting the buy credits flow

My bad @erikalogie, see updated testing instructions please.

> @r41ph looks like you didn't provide the right deploy preview url in the testing instructions, it should be tested on https://deploy-preview-2521--regen-marketplace.netlify.app/

Thanks for spotting that.

**Review by blushi — APPROVED** [review]
utACK
pending conflicts resolution

**Pull request merged**